models/bamf_nnunet_ct_liver/utils/BamfProcessorRunner.py

```diff
@@ -0,0 +1,79 @@
+"""
+-------------------------------------------------
+MedicalHub - Run Module for ensembling nnUNet inference.
+-------------------------------------------------
+-------------------------------------------------
+Author: Rahul Soni
+Email:  rahul.soni@bamfhealth.com
+-------------------------------------------------
+"""
+
+from mhubio.core import Instance, InstanceData, DataType, FileType, CT, SEG
+from mhubio.core import Module, IO
+import os, numpy as np
+import SimpleITK as sitk
+from skimage import measure, filters
+import numpy as np
+import shutil
+
+
+
+class BamfProcessorRunner(Module):
+
+    @IO.Instance
+    @IO.Input('in_data', 'nifti:mod=ct|mr', the='input data to run nnunet on')
+    def task(self, instance: Instance, in_data: InstanceData, out_data: InstanceData) -> None:
+
+       # Log bamf runner info
+        self.v("Running BamfProcessor on....")
+        self.v(f" > input data:  {in_data.abspath}")
+        self.v(f" > output data: {out_data.abspath}")
+
+        # read image
+        self.v(f"Reading image from {in_data.abspath}")
+        img_itk = sitk.ReadImage(in_data.abspath)
+        img_np = sitk.GetArrayFromImage(img_itk)
+
+        # apply post-processing
+        img_bamf_processed = self.n_connected(img_np)
+
+        # store image temporarily
+        out_file = os.path.join(instance.abspath, f'bamf_processed.nrrd')
+        self.v(f"Writing tmp image to {out_file}")
+        img_bamf_processed_itk = sitk.GetImageFromArray(img_bamf_processed)
+        img_bamf_processed_itk.CopyInformation(img_itk)
+        sitk.WriteImage(img_bamf_processed_itk, out_file)
+
+        # meta
+        meta = {
+            "model": "BamfProcessor"
+        }
+
+        # create output data
```

Comment:
Here the `IO.Output` decorator should be used (see [here](https://github.com/MHubAI/documentation/blob/main/documentation/mhubio/how_to_write_an_mhubio_module.md#defining-the-output-data-of-the-module)).

After line 24, add something like this. Please update the explanation of the BAMF Processor accordingly in the `the="..."` parameter.
```python
@IO.Output('out_data', 'nrrd:mod=seg:processor=bamf', data='in_data', the="please explain what the BamfProcessor is doing with the image during the post-processing here")
```

There will be no need to store the image temporarily. 
You already have a output variable defined in the method (`out_data: InstanceData`).

So you'll simply write into that (update  line 45 as follows): 
```Python
sitk.WriteImage(img_bamf_processed_itk, out_data.abspath)
```

Lines 47-56 then can be removed. The confirmation is done automatically by our decorating function if the file is created properly.

Please transfer these changes to the other PR's too.

Reply:
I saw the update, nice! Looks super clean now :)

Reply:
@LennyN95 , thanks for the insights. I have made the necessary changes and updated all four MRs. I will get to you about the test data shortly